fees-margins-estimation/wallet-helper.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"strings"
+)
+
+type WalletConfig struct {
+	URL        string `json:"URL"`
+	Passphrase string `json:"passphrase"`
+	Name       string `json:"Name"`
+}
+
+type Token struct {
+	Token string `json:"token"`
+}
+
+type Keys struct {
+	Keys []struct {
+		Pub     string `json:"pub"`
+		Algo    string `json:"algo"`
+		Tainted bool   `json:"tainted"`
+		Meta    []struct {
+			Key   string `json:"key"`
+			Value string `json:"value"`
+		} `json:"meta"`
+	} `json:"keys"`
+}
+
+var chars = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+func randSeq(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		v, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		b[i] = chars[v.Int64()]
+	}
+	return string(b)
+}
+
+func CheckUrl(url string) bool {
+	return url != "" && (strings.HasPrefix(url, "https://") || strings.HasPrefix(url, "http://"))
+}
+
+func CheckWalletUrl(url string) string {
+	suffix := []string{"/api/v1/", "/api/v1", "/"}
+	for _, s := range suffix {
+		if strings.HasSuffix(url, s) {
+			fmt.Println("There's no need to add ", s, " to WALLETSERVER_URL. Removing it.")
+			return url[0 : len(url)-len(s)]
+		}
+	}
+	return url
+}
+
+func CreateWallet(config WalletConfig) ([]byte, error) {
+	// __create_wallet:
+	// Create a new wallet:
+	jsonStr := []byte("{\"wallet\":\"" + config.Name + "\",\"passphrase\":\"" + config.Passphrase + "\"}")
+	req, err := http.NewRequest("POST", config.URL+"/api/v1/wallets", bytes.NewBuffer(jsonStr))
```

Comment:
Suggestion: Avoid magic strings: `"POST"` --> `http.MethodPost` ([ref](https://golang.org/pkg/net/http/#pkg-constants))

Reply:
I fixed it.